### PR TITLE
Table writer code cleanup

### PR DESCRIPTION
**Pull request by xiaoxmeng**

### CODE REVIEW

File: velox/connectors/hive/HiveDataSink.h [hunk 1]
```diff
@@ -207,7 +208,7 @@ class HiveDataSink : public DataSink {
   const CommitStrategy commitStrategy_;
   // Parameters used by writers, and thus are tracked in the same order
   // as the writers_ vector
-  std::vector<std::shared_ptr<HiveWriterInfo>> writerInfo_;
+  std::vector<std::shared_ptr<HiveWriterInfo>> writerInfos_;
```
Comment (gggrace14) [inline velox/connectors/hive/HiveDataSink.h:211]: Plural of information?

File: velox/connectors/hive/HiveDataSink.cpp [hunk 2]
```diff
@@ -37,12 +37,12 @@ std::string makeUuid() {
 }
 
 std::string makePartitionDirectory(
-    const std::string& tableDirectory,
-    const std::optional<std::string>& partitionSubdirectory) {
-  if (partitionSubdirectory.has_value()) {
-    return (fs::path(tableDirectory) / partitionSubdirectory.value()).string();
+    const std::string& tableDir,
```
Comment (gggrace14) [inline velox/connectors/hive/HiveDataSink.cpp:40]: Presto Java codebase discourages abbrev. Not sure about Velox.

File: velox/connectors/hive/HiveDataSink.cpp [hunk 3]
```diff
@@ -155,11 +154,11 @@ std::shared_ptr<const HiveWriterParameters> HiveDataSink::getWriterParameters(
       break;
     }
     default:
-      VELOX_UNREACHABLE();
+      VELOX_UNREACHABLE(commitStrategyToString(commitStrategy_));
```
Comment (gggrace14) [inline velox/connectors/hive/HiveDataSink.cpp:157]: Guess no need for an explicit value at least with VELOX_UNREACHABLE(). Sounds contradicting.

**Review by gggrace14 — CHANGES_REQUESTED** [review]
This refactoring PR seems to mix refactoring on several independent components. Would you be able to break it down into separate PRs.
Also a general title "some code refactoring" is not informative, or not useful if we trace back in commit history to find a related PRs. Not being able to be specific might also indicate we need to break it down.

File: velox/connectors/hive/HiveDataSink.cpp (hunk 3 above)
Comment (xiaoxmeng) [reply velox/connectors/hive/HiveDataSink.cpp:157]: I think it is helpful to log out the actual value in this case.

File: velox/connectors/hive/HiveDataSink.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/connectors/hive/HiveDataSink.cpp:40]: tableDir is common abbreviation in c++ data system code base.

**Comment by xiaoxmeng** [comment]
> This refactoring PR seems to mix refactoring on several independent components. Would you be able to break it down into separate PRs. Also a general title "some code refactoring" is not informative, or not useful if we trace back in commit history to find a related PRs. Not being able to be specific might also indicate we need to break it down.

It is a very small cleanup diff. Will split.

File: velox/connectors/hive/HiveDataSink.h (hunk 1 above)
Comment (xiaoxmeng) [reply velox/connectors/hive/HiveDataSink.h:211]: It is ok to name like this in variable naming such as groupingKeyInfos_.

File: velox/connectors/hive/HiveDataSink.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/connectors/hive/HiveDataSink.cpp:40]: Ok. Also comment from @oerling "We hav generally preferred directory over dir." Changes in policy? I revert this change.

File: velox/connectors/Connector.cpp [hunk 4]
```diff
@@ -86,7 +86,7 @@ void Connector::unregisterTracker(cache::ScanTracker* tracker) {
 std::shared_ptr<cache::ScanTracker> Connector::getTracker(
     const std::string& scanId,
     int32_t loadQuantum) {
-  return trackers_.withWLock([&](auto& trackers) -> auto {
+  return trackers_.withWLock([&](auto& trackers) -> auto{
```
Comment (mbasmanova) [inline velox/connectors/Connector.cpp:89]: Why remove the space after 'auto'? BTW, why do we need -> auto at all?

File: velox/connectors/Connector.cpp [hunk 5]
```diff
@@ -111,7 +111,8 @@ std::string commitStrategyToString(CommitStrategy commitStrategy) {
     case CommitStrategy::kTaskCommit:
       return "TASK_COMMIT";
     default:
-      VELOX_UNREACHABLE();
+      return fmt::format(
```
Comment (mbasmanova) [inline velox/connectors/Connector.cpp:114]: What is the motivation for this change?

File: velox/connectors/hive/HiveDataSink.cpp [hunk 6]
```diff
@@ -155,11 +154,11 @@ std::shared_ptr<const HiveWriterParameters> HiveDataSink::getWriterParameters(
       break;
     }
     default:
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveDataSink.cpp:156]: Do we need the 'default' clause given that we handled all the enum values?

**Review by mbasmanova — COMMENTED** [review]
@xiaoxmeng Some questions.

File: velox/connectors/Connector.cpp (hunk 5 above)
Comment (xiaoxmeng) [reply velox/connectors/Connector.cpp:114]: It is good to know the exact bad enum number when this happens?

File: velox/connectors/Connector.cpp (hunk 4 above)
Comment (xiaoxmeng) [reply velox/connectors/Connector.cpp:89]: clang format does so. Let me double check.
